### DISCUSSION TITLE
Replace Blackhole?

### DISCUSSION THREAD

**Discussion by darwinDMG** [kyleneideck/BackgroundMusic#428] [Q&A]
Hello!

I'm planning on using this app to control volume on different apps (like muting a Zoom call) but I also notice that it can route sounds between applications, ys? If so would this replace a plugin like Blackhole? I'm currently using that app to send audio between applications like OBS. 

JVK

**Comment by ManuelHeche1983** [comment]
Hey! I also want to use Blackhole for OBS to Zoom (Mic and Guitar through NI Komplete 6 and Desktop Audio) Maybe this one is the better solution, but may I ask, what your settings are to get your things to work with Blackhole?

**Reply by darwinDMG** [reply]
In my case, I've had to set both the Input and Output to Blackhole in the system Sound prefs, and then the audio gets patched through. You also have to make sure you've set up Blackhole correctly as a new device in the MIDI controls (per the install instructions) for this to work.

**Comment by kyleneideck** [comment]
I haven't used BlackHole or OBS, but if OBS just needs a simple loopback driver, BGMDriver (the Background Music driver) should work fine as far as I know. Have you tried it?

To minimise latency, I think you can add the Background Music device to an aggregate or multi-output device in a similar way to the process described in the instructions for BlackHole. The Background Music app only doesn't set it up that way because multi-output devices don't support controls (volume, mute, etc) yet. If you don't mind only being able to control your volume through System Preferences, it's better to use a multi-output device. (Except that apparently multi-output devices are "[pretty buggy](https://github.com/ExistentialAudio/BlackHole#what-apps-dont-work-with-multi-outputs)".)

Depending on how complex your setup is, you might run into limitations like BGMDriver only currently supporting 2-channel devices. It also doesn't have a way to configure it to publish multiple input or output devices, so you can't use it to, for example, route the audio from app A to app B while also routing the audio from app C to app D. Actually, I'm not sure if BlackHole supports that either.
